### PR TITLE
Pipefix july

### PR DESCRIPTION
**Pull request by jxyang**
This PR fixes hangs in the event of shutdown for two cases:

* dotnet-proc-maps
* libcxx when a child thread fails an assertion

In both cases, there were thread waiting for a futex with infinite timeout, thus caused the process not able to shutdown.  This PR attempts to wake up the futex while sending the kill signal to the thread.

### CODE REVIEW

File: kernel/thread.c [hunk 1]
```diff
@@ -1242,7 +1247,14 @@ size_t myst_kill_thread_group()
         for (t = tail; t != NULL; t = t->group_prev)
         {
             if (t != process && t != thread && t->status != MYST_ZOMBIE)
+            {
```
Comment (paulcallen) [inline kernel/thread.c:1250]: may wont to delete the printf!
In my PR I did another WAKE here as there is a race condition between the signal on the killing and the waiting. The code may be in a transitional state where it has not quite made it to the wait and so the previous wake may not be enough. Adding it here also gets that other race.

File: kernel/thread.c [hunk 2]
```diff
@@ -1210,14 +1210,19 @@ size_t myst_kill_thread_group()
...
-        //        if (!myst_is_process_thread(t) && t->status == MYST_RUNNING)
         if (t != thread && t->status == MYST_RUNNING)
         {
             count++;
             myst_spin_unlock(process->thread_lock);
             myst_signal_deliver(t, SIGKILL, 0);
+
             // Wake up the thread from futex_wait if necessary.
-            myst_cond_signal_thread(t->signal.cond_wait, t);
+            if (t->signal.waiting_on_event)
```
Comment (vtikoo) [inline kernel/thread.c:1220]: Can we move this to `myst_signal_deliver`? That would cover waking futexes in non-shutdown scenarios, like musl SIGSYNCCALL.
However, in the non-shutdown case there is also the issue of restarting syscalls if SA_RESTART is set.

File: kernel/thread.c (hunk 1 above)
Comment (jxyang) [reply kernel/thread.c:1250]: Will add this to the following loop. Thanks.

File: kernel/thread.c (hunk 2 above)
Comment (jxyang) [reply kernel/thread.c:1220]: The last time I tried that, dotnet-proc-maps hangs for some reasons. I will give it another try.

I am not `SA_RESTART` applies when a kill signal interrupts the operation.

**Review by paulcallen — APPROVED** [review]

**Pull request merged**